measure_cocoapod_size.py

```diff
@@ -96,6 +106,18 @@ def CopyProject(source_dir, target_dir):
   """
   os.system('cp -r {} {}'.format(source_dir, target_dir))
 
+def ValidateSourceConfig(pod_sources):
+  for sdk , source in pod_sources.items():
+    if len(source) >= 1 and ( source.keys()[0] not in set(["git", "path"]) ):
+      raise ValueError(
+              "Pod source of SDK {} should be `git` or `path`.".format(sdk))
+    elif len(source) > 1:
+      if source.keys()[0] != "git":
+        raise ValueError(
+                "For multiple specs for the SDK {} ,`git` should be added with `branch`, `tag` or `commit`".format(sdk))
+      if source.keys()[1] not in set(["branch", "tag", "commit"]):
+        raise ValueError(
+                "A specified version of the SDK {} should be from `branch`, `tag` or `commit`.".format(sdk))
```

Comment:
Hmm I think something could go wrong if the config file looked like:
<details>
<summary>Invalid JSON</summary>

```bash
      {
        "Alamofire":
        {
          "git": "https://github.com/Alamofire/Alamofire.git",
          "branch": "dev",
          "commit": "0f506b1c45" ← This makes the config invalid but I think it would be falsely validated.
        }
      }
```
</details>

So maybe start the loop logic with two checks:
```python
if len(source.keys()) > 2:
  raise ValueError(
    "Pod source of SDK {} can only specify `path`, `git`, or `git` and a reference (like a `branch`, `commit`, or `tag`)."
    "See --help for an example config."
    .format(sdk)
  )
```